stellargraph/data/explorer.py

```diff
@@ -233,6 +231,35 @@ def _check_sizes(self, n_size):
             if type(d) != int or d < 0:
                 self._raise_error(err_msg)
 
+    def _sample_neighbours_untyped(
+        self, neigh_func, py_and_np_rs, cur_node, size, weighted
+    ):
+        """
+        Sample ``size`` neighbours of ``cur_node`` without checking node types or edge types, optionally
+        using edge weights.
+        """
+        if cur_node != -1:
+            neighbours = neigh_func(
+                cur_node, use_ilocs=True, include_edge_weight=weighted
+            )
+
+            if weighted:
+                neighbours, weights = neighbours
+        else:
+            neighbours = []
+
+        if len(neighbours) == 0:
```

Comment:
I wonder if we should do something like:
```python
if len([w for w in weights if w != 0]) == 0:
```
so that we still have `-1`s propagated if all neighbours have zero weight?

Reply:
Hm, that's a good point. That particular form is likely to be expensive because it's manually iterating the array, but it's probably worth doing something with https://docs.scipy.org/doc/numpy/reference/generated/numpy.count_nonzero.html or using the final value of the `cumsum` in `naive_weighted_choice`. I'll probably do it in the latter, because then it applies to weighted Node2Vec too (which currently doesn't handle this case).

Reply:
Done, and fixed/tested in Node2Vec and CTDNE as appropriate as well.